### ISSUE TITLE
Cannot launch aws_instance using an AMI getting an error

### ISSUE THREAD

**Issue by tmtk75** [hashicorp/terraform#2225]
This is the `.tf` file I tried. The region is `ap-northeast-1`.
```
resource "aws_instance" "foo" {
    ami = "ami-81294380"
    instance_type = "m3.large"
    key_name = "my-key"
    subnet_id = "subnet-dd3c97aa"
    root_block_device {
        volume_size = 20
    }
}
```
ami-81294380 is https://aws.amazon.com/marketplace/pp?sku=eggbgx9svw4xhzs1omttdv29q.

I got following error when I ran apply.
```
$ terraform apply
aws_instance.foo: Creating...
  ami:                                       "" => "ami-81294380"
  availability_zone:                         "" => "<computed>"
  ebs_block_device.#:                        "" => "<computed>"
  ephemeral_block_device.#:                  "" => "<computed>"
  instance_type:                             "" => "t1.micro"
  key_name:                                  "" => "my-key"
  placement_group:                           "" => "<computed>"
  private_dns:                               "" => "<computed>"
  private_ip:                                "" => "<computed>"
  public_dns:                                "" => "<computed>"
  public_ip:                                 "" => "<computed>"
  root_block_device.#:                       "" => "1"
  root_block_device.0.delete_on_termination: "" => "1"
  root_block_device.0.iops:                  "" => "<computed>"
  root_block_device.0.volume_size:           "" => "20"
  root_block_device.0.volume_type:           "" => "<computed>"
  security_groups.#:                         "" => "<computed>"
  subnet_id:                                 "" => "subnet-dd3c97aa"
  tenancy:                                   "" => "<computed>"
  vpc_security_group_ids.#:                  "" => "<computed>"
aws_instance.foo: Error: 1 error(s) occurred:

* Error launching source instance: InvalidBlockDeviceMapping: logicalType cannot be modified on root device
        status code: 400, request id: []
Error applying plan:

1 error(s) occurred:

* 1 error(s) occurred:

* 1 error(s) occurred:

* Error launching source instance: InvalidBlockDeviceMapping: logicalType cannot be modified on root device
        status code: 400, request id: []

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

If I removed next `root_block_device`, I was able to launched properly.
```
    root_block_device {
        volume_size = 20
    }
```

If I changed AMI to `ami-cbf90ecb` (Amazon Linux of ap-northeast-1) with `root_block_device`, It was also launched.
```
    ...
    ami = "ami-cbf90ecb"
    ...
    root_block_device {
        volume_size = 20
    }
```

**Comment by tmtk75** [comment]
Actually `block_device` of `0.3.7` worked to change volume size of root device.

```
resource "aws_instance" "foo" {
    ami = "ami-81294380"
    ....
    block_device {
        device_name = "/dev/sda"
        volume_size = 20
    }
}
```

And after launching with 0.3.7, I could resize with `resize2fs`.
```
[root@ip-172-31-14-229 ~]# resize2fs /dev/xvde
resize2fs 1.41.12 (17-May-2010)
Filesystem at /dev/xvde is mounted on /; on-line resizing required
old desc_blocks = 1, new_desc_blocks = 2
Performing an on-line resize of /dev/xvde to 5242880 (4k) blocks.
The filesystem on /dev/xvde is now 5242880 blocks long.

[root@ip-172-31-14-229 ~]# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/xvde        20G  654M   19G   4% /
tmpfs           296M     0  296M   0% /dev/shm
```

**Comment by tmtk75** [comment]
Here is a .tf file to reproduce. I tried with v0.5.3.
https://gist.github.com/tmtk75/9d871f734a0597107f67

**Comment by joshgarnett** [comment]
@tmtk75 does it work if you specify the volume type?

```
  root_block_device {
    volume_size = 20
    volume_type = "gp2"
  }
```

**Comment by phinze** [comment]
Hi @tmtk75 - this is an expression of a bug that is fixed in #2271.

Notice how the in the definition of the AMI you're using, the RootBlockDevice does not match anything in the list of BlockDeviceMappings:

```js
{
    "Images": [
        {
            "ProductCodes": [
                {
                    "ProductCodeId": "eggbgx9svw4xhzs1omttdv29q",
                    "ProductCodeType": "marketplace"
                }
            ],
            "Name": "CentOS-6.5-GA-03.3-f4325b48-37b0-405a-9847-236c64622e3e-ami-6be4dc02.2",
            "VirtualizationType": "paravirtual",
            "Hypervisor": "xen",
            "ImageOwnerAlias": "aws-marketplace",
            "ImageId": "ami-81294380",
            "RootDeviceType": "ebs",
            "State": "available",
            "BlockDeviceMappings": [
                {
                    "DeviceName": "/dev/sda",
                    "Ebs": {
                        "DeleteOnTermination": false,
                        "SnapshotId": "snap-50d7a3be",
                        "VolumeSize": 8,
                        "VolumeType": "standard",
                        "Encrypted": false
                    }
                }
            ],
            "Architecture": "x86_64",
            "ImageLocation": "aws-marketplace/CentOS-6.5-GA-03.3-f4325b48-37b0-405a-9847-236c64622e3e-ami-6be4dc02.2",
            "KernelId": "aki-44992845",
            "OwnerId": "679593333241",
            "RootDeviceName": "/dev/sda1",
            "CreationDate": "2014-02-05T16:53:07.000Z",
            "Public": true,
            "ImageType": "machine",
            "Description": "CentOS-6.5-GA-03.3 on EBS x86_64 20140204:2336"
        }
    ]
}
```

The good news is that this should work on master and the forthcoming 0.6 release! :+1:

**Comment by tmtk75** [comment]
Hi @phinze, my code at the top worked with 0.6.0! Thank you!
